File: for-local-test/python/414.third-maximum-number.py
class Solution:
    def thirdMax(self, nums: list[int]) -> int:
        unique_nums = set(nums)
        if len(unique_nums) < 3:
            return max(unique_nums)
        else:
            sorted_nums = sorted(unique_nums, reverse=True)
            return sorted_nums[2]
        
if __name__ == "__main__":
    solution = Solution()
    nums = [3, 2, 1]
    result = solution.thirdMax(nums)
    print(result)  # Output: 1

File: for-local-test/414.third-maximum-number.java
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

class Solution {
    public int thirdMax(int[] nums) {
        List<Integer> list = new ArrayList<>();
        for(int num : nums){
            if(!list.contains(num)){
                list.add(num);
            }
        }
        if(list.size() < 3){
            return Collections.max(list);
        }
        Collections.sort(list);
        return list.get(list.size() - 3);
    }
    public static void main(String[] args) {
        Solution solution = new Solution();
        int[] nums = {3, 2, 1};
        int result = solution.thirdMax(nums);
        System.out.println(result); // Output: 1
    }
}


